### PR TITLE
Android13 QPR2 Dependency & Misc Updates

### PR DESCRIPTION
**Pull request by theimpulson**
This PR updates dependencies to the latest release (as per AOSP and Gradle) wherever possible.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,11 +1,11 @@
 buildscript {
     // 1.3.61 Android 11
     // 1.4.30 Android 12
-    // 1.6.10 Android 13
```
Comment (chirayudesai) [inline build.gradle:4]: Nit: I'd leave this lin in

File: build.gradle [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+    // 1.7.20 Android 13 (TQ2A)
```
Comment (chirayudesai) [inline build.gradle]: It's usually called QPR2

**Review by chirayudesai — DISMISSED** [review]
We could also rebuild `android.jar` and `libcore.jar`

File: build.gradle (hunk 1 above)
Comment (theimpulson) [reply build.gradle:4]: Done

File: build.gradle (hunk 2 above)
Comment (theimpulson) [reply build.gradle]: Thanks, changed

**Comment by chirayudesai** [comment]
I'd like to do a bit of quick testing with this but otherwise it all looks good.

**Comment by grote** [comment]
yeah I'd be most worried about the room upgrade to not affect our K/V DB updates.

**Comment by theimpulson** [comment]
The libs versions seem to be the same in the latest QPR3

**Comment by theimpulson** [comment]
Rebased to fix conflicts

File: storage/lib/Android.bp [hunk 3]
```diff
@@ -40,7 +40,7 @@ android_library {
         enabled: false,
     },
     kotlincflags: [
-        "-Xopt-in=kotlin.RequiresOptIn",
+        "-opt-in=kotlin.RequiresOptIn",
```
Comment (grote) [inline storage/lib/Android.bp:43]: Is this how they do it now? Is that documented somewhere?

File: gradle/dependencies.gradle [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ext {
-    // https://android.googlesource.com/platform/prebuilts/sdk/+/refs/tags/android-13.0.0_r3/current/androidx/Android.bp#3901
-    ext.room_version = "2.4.0-alpha05"
-    // https://android.googlesource.com/platform/external/protobuf/+/refs/tags/android-13.0.0_r3/java/pom.xml#7
+    // https://android.googlesource.com/platform/prebuilts/sdk/+/refs/tags/android-13.0.0_r32/current/androidx/Android.bp#4586
+    ext.room_version = "2.5.0-alpha01"
```
Comment (grote) [inline gradle/dependencies.gradle]: Too bad they include an early alpha for something so critical.

File: storage/lib/Android.bp (hunk 3 above)
Comment (theimpulson) [reply storage/lib/Android.bp:43]: This was hinted at in a compile-time warning. As I see it is mentioned here in docs: https://kotlinlang.org/docs/opt-in-requirements.html#module-wide-opt-in

File: gradle/dependencies.gradle (hunk 4 above)
Comment (theimpulson) [reply gradle/dependencies.gradle]: Does this matter, as it has been used in AOSP for quite a while?

Comment (chirayudesai) [reply gradle/dependencies.gradle]: It's not used extensively in AOSP.

See https://cs.android.com/search?q=androidx.room_room-runtime&start=1

From what we know, Google mostly checks in prebuilts.

@grote Happy to just check in a prebuilt from this in a separate change (and anything else that we don't want to rely on AOSP versions for - if they version is too old, or alpha, or has any other issue).

Comment (theimpulson) [reply gradle/dependencies.gradle]: Removed for unblocking it, added a comment instead

**Comment by theimpulson** [comment]
Bumped AGP and Gradle to 8.x, ktlint to latest stable version and other changes to accommodate them

**Comment by grote** [comment]
So this forces me to use JDK 17 and AS Giraffe I guess :/

> The project is using an incompatible version (AGP 8.1.0) of the Android Gradle plugin. Latest supported version is AGP 8.0.2

File: gradle/dependencies.gradle (hunk 4 above)
Comment (grote) [reply gradle/dependencies.gradle]: Thanks for removing. I thought about it and actually there's probably no point in removing it. We build this app with AOSP, so whatever is in AOSP gets used. Removing this here just affects our gradle builds which are dev only.

So, the testing would actually need to happen when AOSP bumps its deps. Ideally, we'd also have some integration tests for K/V backups with room. E.g. Some sample data in the repo and we try to read and compare the result with what we expect. Then just running these tests (on-device!) would already give us more peace of mine.

**Comment by grote** [comment]
So after upgrading my local setup, building it and running tests works here. However, there's an issue with the gradle build light theme where the action bar is all black including the text.

**Review by grote — APPROVED** [review]

**Comment by theimpulson** [comment]
> So after upgrading my local setup, building it and running tests works here. However, there's an issue with the gradle build light theme where the action bar is all black including the text.

@grote it seems to be unrelated to this MR. I tried looking and it seems the aforementioned issue is coming from the fact that colors defined for light theme [here](https://github.com/seedvault-app/seedvault/blob/android13/app/src/main/res/values/colors.xml#L7) are using system colors while dark theme colors [here](https://github.com/seedvault-app/seedvault/blob/android13/app/src/main/res/values-night/colors.xml#L7) simply use a properly defined color. I am unsure why it was done that way but maybe we can handle that in a different MR?

File: gradle/dependencies.gradle (hunk 4 above)
Comment (theimpulson) [reply gradle/dependencies.gradle]: Opened an issue to track things here: https://github.com/seedvault-app/seedvault/issues/548

**Comment by theimpulson** [comment]
> > So after upgrading my local setup, building it and running tests works here. However, there's an issue with the gradle build light theme where the action bar is all black including the text.
> 
> @grote it seems to be unrelated to this MR. I tried looking and it seems the aforementioned issue is coming from the fact that colors defined for light theme [here](https://github.com/seedvault-app/seedvault/blob/android13/app/src/main/res/values/colors.xml#L7) are using system colors while dark theme colors [here](https://github.com/seedvault-app/seedvault/blob/android13/app/src/main/res/values-night/colors.xml#L7) simply use a properly defined color. I am unsure why it was done that way but maybe we can handle that in a different MR?

Issue to keep track: https://github.com/seedvault-app/seedvault/issues/549

**Pull request merged**